### ISSUE TITLE
TreeSelect component should support 'aria-label' option prop.

### ISSUE THREAD

**Issue by benbenbuhben** [ant-design/ant-design#40055]
### What problem does this feature solve?

The <Select /> component supports an 'aria-label' option prop, which is useful when the label prop is a React component (and therefore cannot be read by the screen reader). However, the <TreeSelect /> component does not seem to support an 'aria-label' prop - instead, the value prop is read by the screen reader in every case. Please reference the following CodeSandbox examples using a screen reader:

[Example #1 - Select component uses 'aria-label' prop as expected](https://codesandbox.io/s/basic-usage-antd-5-1-2-forked-fv0zgh?file=/demo.tsx)

[Example #2 - TreeSelect does not use 'aria-label' prop](https://codesandbox.io/s/g5wpqd?file=/demo.tsx)

### What does the proposed API look like?

Add 'aria-label' as a supported option property on tree nodes.

**Comment by guan404ming** [comment]
I'll try to work on this!!

**Comment by benbenbuhben** [comment]
@afc163 - This was closed as completed, but even after upgrading to antd@5.7.3, my example Code Sandbox for the TreeSelect still has the problem of the screenreader reading the 'value' prop instead of the 'aria-label' prop. It seems that the aria-label prop has no effect. It does still work as expected in the Select component, just not in the TreeSelect. Should this be re-opened?

[Example - TreeSelect does not use 'aria-label' prop (SHOULD BE FIXED)](https://codesandbox.io/s/basic-antd-5-7-3-forked-pn6tyc?file=/demo.tsx)

**Comment by benbenbuhben** [comment]
I have investigated this issue further. On inspecting the DOM, I found that the aria-label attribute is being correctly applied to the tree nodes that have an aria-label prop passed to them. However, these aria-labels are not being announced by the screen reader (I'm using VoiceOver on macOS). This seems to be due to a span element within the dropdown that has aria-live="assertive" applied. This span is set to always read the value prop of the active node, even when the node has an aria-label prop.

The relevant code appears to be in the rc-tree-select repository:

https://github.com/react-component/tree-select/blob/master/src/OptionList.tsx#L213-L215

Should I raise an issue in that repository instead?

**Comment by afc163** [comment]
@benbenbuhben Sure.
